ui/components/MeshModelRegistry/MeshModelComponent.js

```diff
@@ -115,174 +114,168 @@ const MeshModelComponent = ({
   const [checked, setChecked] = useState(true);
   // const [loading, setLoading] = useState(false);
 
-  useEffect(() => {
-    if (selectedTab && selectedTab !== OVERVIEW) {
-      setAnimate(true);
-      setConvert(true);
-      setView(selectedTab);
-    }
-  }, [selectedTab]);
+  /**
+   * RTK Lazy Queries
+   */
+  const [getMeshModelsData] = useLazyGetMeshModelsQuery();
+  const [getComponentsData] = useLazyGetComponentsQuery();
+  const [getRelationshipsData] = useLazyGetRelationshipsQuery();
+  const [getRegistrantsData] = useLazyGetRegistrantsQuery();
 
-  const getModels = async (page) => {
+  const fetchData = useCallback(async () => {
```

Comment:
Reason of using useCallback?

Reply:
It will make sure we are not making unintentional API calls, however RTK do support cache which but this function is update states also, useCallback will make sure to not have any unintentional behaviour due to state change